chain/client/src/client_actor.rs

```diff
@@ -662,6 +664,50 @@ impl Handler<Status> for ClientActor {
                 earliest_block_time = Some(earliest_block.timestamp());
             }
         }
+        // Provide more detailed information about the current state of chain.
+        // For now - provide info about last 50 blocks.
+        let detailed_debug_status = if msg.get_detailed_info {
+            let mut blocks_debug: Vec<DebugBlockStatus> = Vec::new();
+
+            let mut last_block_hash = head.last_block_hash;
+            let mut last_block_timestamp: u64 = 0;
+
+            // Fetch last 50 blocks (we can fetch more blocks in the future if needed)
+            for _ in 0..50 {
+                let block = match self.client.chain.get_block(&last_block_hash) {
+                    Ok(block) => block,
+                    Err(_) => break,
+                };
+                let chunks = block
+                    .chunks()
+                    .iter()
+                    .map(|chunk| DebugChunkStatus {
+                        shard_id: chunk.shard_id(),
+                        chunk_hash: chunk.chunk_hash(),
+                        gas_used: chunk.gas_used(),
+                        processing_time_ms: CryptoHashTimer::get_timer_value(chunk.chunk_hash().0),
+                    })
+                    .collect();
+
+                blocks_debug.push(DebugBlockStatus {
+                    block_hash: last_block_hash,
+                    block_height: block.header().height(),
+                    chunks,
+                    processing_time_ms: CryptoHashTimer::get_timer_value(last_block_hash),
+                    timestamp_delta: if last_block_timestamp > 0 {
```

Comment:
If I understand correctly, `timestamp_delta` is the time difference between the timestamp of this block and the next block. I think this could be confusing. Intuitively, when I look at the stats, I would assume that this number is the timestamp between the previous block and this block. 

I understand implementation wise, the current way is easier. But this may cause confusion when looking at the debug page number

Reply:
yes, this is easier implementation wise, but I agree that it might be confusing in the current HTML.

very soon, I plan to update that HTML (adding colors etc) - and it should be more clear for the user.